pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -288,6 +263,20 @@ func (e *Controller) onServiceDelete(obj interface{}) {
 	e.queue.Add(key)
 }
 
+// onPodUpdate enqueues the pod's projection key on Add/Update/Delete events, to find matching services later.
+func (e *Controller) onPodUpdate(old, cur interface{}) {
+	var key *endpointsliceutil.PodProjectionKey
+	if old == nil {
+		key = endpointsliceutil.GetPodProjectionKey(cur)
```

Comment:
Why do we need `GetPodProjectionKey` at all?

Why can't we use `GetPodUpdateProjectionKey` for all combinations of old, cur?

Reply:
changed it